Merge 6LoWPAN tutorials (#165)
micro-ROS/micro-ROS.github.io: _docs/tutorials/advanced/nuttx/6lowpan/index.md

---
title: 6LoWPAN Guide
permalink: /docs/tutorials/advanced/nuttx/6lowpan/
---

In this guide, we will show how to use micro-ROS over 6LoWPAN communication.

# What is 6LoWPAN?

6LoWPAN is an acronym o IPv6 over Low-Power Wireless Personal Area Networks.
This communication protocol allows wireless communication over IEEE 802.15.4 based networks using IPv6. Some of the main advantages are:
- Easy to route from radio devices to the Internet, thanks to the usage of the IP packets.

> - Easy to use on UDP and TCP server/clients.j

Reviewer: ```suggestion
- Easy to use on UDP and TCP server/clients.
```

- A protocol designed for low power and constrained devices, perfect or micro-ROS remote sensors.

# Needed hardware

> At this moment, 6LoWPAN in only available for NuttX RTOS.

Reviewer: ```suggestion
At present, 6LoWPAN is only available for the NuttX RTOS.
```

> You need the following devices to implement the steps highlighted in this guide:

Reviewer: ```suggestion
In order to implement the steps highlighted in this guide, you need the following devices:
```

Builtin Apps:
  ping6         i8sak         uros_6lowpan 
```
- Now turn-on the RPi and execute the micro-ROS-bridge-RPI tool by typing the next command:

```bash
./ ~/micro-ROS-HB.sh
```
- Once everything is configured, it will return the connection data of the 6LoWPAN network:

```bash
lowpan0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1280
        inet6 fe80::b482:ca65:743b:b6bd  prefixlen 64  scopeid 0x20<link>
        unspec B6-82-CA-65-74-3B-B6-BD-00-00-00-00-00-00-00-00  txqueuelen 1000  (UNSPEC)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 21  bytes 2242 (2.1 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

1) Add new 6LoWPAN micro-ROS device	 3) Create UDP micro-ROS Agent		  5) Create Serial micro-ROS Agent server
2) Create UDP 6LoWPAN micro-ROS Agent	 4) Create TCP micro-ROS Agent		  6) Quit
#? 
```
- The value **inet6** is the IPv6 direction of the RPi over the 6LoWPAN interface. Copy it because is necessary for the subsequent steps.
- Execute the micro-ROS 6LoWPAN application on the Olimex typing the next command:

```bash
uros_6lowpan <Agent_IP> <Agent_Port> <pub/sub>
```

>   where `Agent_IP`is IPv6 copying previously, `Agent_PORT` is the port selected at Agent and `pub/sub` controls the application behavior, in case of `pub` it will behalf as publisher and as a subscriber in `sub` option.

Reviewer: ```suggestion
  where `Agent_IP` is the IPv6 copied previously, `Agent_PORT` is the port selected for the Agent and `pub/sub` controls the application behavior: in case of `pub` it will act as publisher and as in case of `sub`, it will act as a subscriber.
```
